skimage/exposure/_adapthist.py

```diff
@@ -13,21 +13,22 @@
 responsible.  Basically, don't be a jerk, and remember that anything free
 comes with no guarantee.
 """
+from __future__ import division
+import numbers
 import numpy as np
 from .. import img_as_float, img_as_uint
 from ..color.adapt_rgb import adapt_rgb, hsv_value
 from ..exposure import rescale_intensity
-from ..util import view_as_blocks, pad
+from ..util import view_as_windows
+from .._shared.utils import skimage_deprecation
 
 
-MAX_REG_X = 16  # max. # contextual regions in x-direction */
-MAX_REG_Y = 16  # max. # contextual regions in y-direction */
 NR_OF_GREY = 2 ** 14  # number of grayscale levels to use in CLAHE algorithm
 
 
 @adapt_rgb(hsv_value)
-def equalize_adapthist(image, ntiles_x=8, ntiles_y=8, clip_limit=0.01,
-                       nbins=256):
+def equalize_adapthist(image, kernel_size=64, ntiles_x=None, ntiles_y=None,
+                       clip_limit=0.01, nbins=256):
```

Comment:
@JDWarner yes, as mentioned below the result can change if it is more correct. (that makes it a bug fix, rather than an API breakage.) I think we *absolutely* should hold fast to our API no-breakage policy. The future API *should* be what you mention (kernel size immediately following the image), but we will have to wait until 0.14 to ship it: two full versions of warning messages for users who haven't updated their arg/kwarg usage.